python/pyarrow/dataset.py

```diff
@@ -203,6 +204,15 @@ def _ensure_partitioning(scheme):
     return scheme
 
 
+def _ensure_format(obj):
+    if isinstance(obj, FileFormat):
+        return obj
+    elif obj == "parquet":
+        return ParquetFileFormat()
+    elif not isinstance(obj, FileFormat):
```

Comment:
```suggestion
    else:
```

you already checked above for FileFormat

Reply:
Updated.